lib/api.js

```diff
@@ -43,12 +46,14 @@ module.exports.methods = {
             throw new TypeError('Settings are not provided');
         }
         
-        if (!settings.apiKey || (settings.apiKey && typeof (settings.apiKey) !== 'string')) {
-            debug.write('You have to pass API key to initialize Stackify Logger');
-            throw new TypeError('You have to pass API key');
+        if (settings.transport != 'agent_socket') {
+            if (!settings.apiKey || (settings.apiKey && typeof (settings.apiKey) !== 'string')) {
+                debug.write('You have to pass API key to initialize Stackify Logger');
+                throw new TypeError('You have to pass API key');
+            }
```

Comment:
I guess it is better to do it in positive way..

```
if (settings.transport == 'agent_socket') {
  // do things
} else {
  // do something else
}
```

in this way if we want to add another transport you just have to add `else if` block instead of refactoring most of the code.